src/Symfony/Component/Serializer/Normalizer/DenormalizerInterface.php

```diff
@@ -34,7 +34,7 @@ interface DenormalizerInterface
      * @param string $format  Format the given data was extracted from
      * @param array  $context Options available to the denormalizer
      *
-     * @return object
+     * @return object|string
```

Comment:
this should be reverted, it's unrelated

Reply:
But actually, this is a problem: this type is the only legal return type, and we could actually enforce it in master with a real return type.
This means this PR will break. Is this a hint this  is not the correct approach?
The denormalization logic should throw an exception to me, but maybe we would add a generic option that would allow the caller to catch the exception and leave the original value in place?

Reply:
You mean, create a new Exception class which will be more specific than the NotNormalizableValueException ?

Reply:
Or catch `NotNormalizableValueException` and leave original values in place when `disable_type_enforcement` is set (or another option if the semantics doesn't fit?)

Reply:
Not related, but maybe that we should allow this method to return mixed. There are valid use cases where you want to “denormalize” a string as another string.
I’ll try the changes introduced in master with API Platform, but I think they’ll break somebody of our normalizers.